src/test/scala/ch/epfl/bluebrain/nexus/tests/admin/AdminSpec.scala

```diff
@@ -0,0 +1,312 @@
+          json shouldEqual jsonContentOf("/admin/errors/not-exists.json", errorCtx)
+        }
+      }
+
+      "succeed and fetch revisions" in {
+        val id     = genId()
+        val create = reqEntity(nxv = "nxv", person = "person")
+        cl(Req(PUT, s"$adminBase/projects/$id", entity = create)).mapJson { (json, result) =>
+          result.status shouldEqual StatusCodes.Created
+          json shouldEqual createRespJson(id, 1L)
+        }
+
+        val updateRev1 = reqEntity("/admin/update.json", "nxv", "person")
+        cl(Req(PUT, s"$adminBase/projects/$id?rev=1", entity = updateRev1)).mapJson { (json, result) =>
+          result.status shouldEqual StatusCodes.OK
+          json shouldEqual createRespJson(id, 2L)
+        }
+
+        val updateRev2 = reqEntity("/admin/update.json", "nxv", "person")
+        updateRev1.dataBytes
+        cl(Req(PUT, s"$adminBase/projects/$id?rev=2", entity = updateRev2)).mapJson { (json, result) =>
+          result.status shouldEqual StatusCodes.OK
+          json shouldEqual createRespJson(id, 3L)
+        }
+
+        cl(Req(uri = s"$adminBase/projects/$id")).mapJson { (json, result) =>
+          result.status shouldEqual StatusCodes.OK
+          json shouldEqual (updateRev2.toJson.value deepMerge entityMeta(id, 3L))
+        }
+
+        cl(Req(uri = s"$adminBase/projects/$id?rev=3")).mapJson { (json, result) =>
+          result.status shouldEqual StatusCodes.OK
+          json shouldEqual (updateRev2.toJson.value deepMerge entityMeta(id, 3L))
+        }
+
+        cl(Req(uri = s"$adminBase/projects/$id?rev=2")).mapJson { (json, result) =>
+          result.status shouldEqual StatusCodes.OK
+          json shouldEqual (updateRev1.toJson.value deepMerge entityMeta(id, 2L))
+        }
+
+        cl(Req(uri = s"$adminBase/projects/$id?rev=1")).mapJson { (json, result) =>
+          result.status shouldEqual StatusCodes.OK
+          json shouldEqual (create.toJson.value deepMerge entityMeta(id, 1L))
+        }
+      }
+    }
+
+    "deprecating a project" should {
+
+      "fail when wrong revision is provided" in {
+        val id = genId()
+        cl(Req(PUT, s"$adminBase/projects/$id", entity = reqEntity())).mapJson { (json, result) =>
+          result.status shouldEqual StatusCodes.Created
+          json shouldEqual createRespJson(id, 1L)
+        }
+
+        cl(Req(DELETE, s"$adminBase/projects/$id?rev=4")).mapJson { (json, result) =>
+          result.status shouldEqual StatusCodes.Conflict
+          json shouldEqual jsonContentOf("/admin/errors/incorrect-revision.json", errorCtx)
+        }
+      }
+
+      "fail when project does not exist" in {
+        cl(Req(DELETE, s"$adminBase/projects/${genId()}?rev=1")).mapJson { (json, result) =>
+          result.status shouldEqual StatusCodes.NotFound
+          json shouldEqual jsonContentOf("/admin/errors/not-exists.json", errorCtx)
+        }
+      }
+
+      "return not found when revision is not provided" in {
```

Comment:
this doesn't seem right; it may return not found because the id is randomly generated

Reply:
I would say it should return Conflict if the revision is incorrect or not provided